lib/news_scraper/configuration.rb

```diff
@@ -0,0 +1,16 @@
+module NewsScraper
+  class Configuration
+    DEFAULT_SCRAPE_PATTERNS_FILEPATH = File.expand_path('../../../config/article_scrape_patterns.yml', __FILE__)
+    attr_accessor :scrape_patterns, :scrape_patterns_filepath
+
+    def initialize(scrape_patterns_filepath: nil)
+      self.scrape_patterns_filepath = scrape_patterns_filepath if scrape_patterns_filepath
```

Comment:
You need to replace this with 
```
@scrape_patterns_filepath = DEFAULT_SCRAPE_PATTERNS_FILEPATH
```

Or else this won't be set at all during initialization.